content/articles/creating-a-shopping-cart-api/index.md

```diff
@@ -0,0 +1,592 @@
+### Cart Package
+cart class
+```java
+@Component
+public class Cart {
+    //The cart is an aggregation of product
+    private List<Product> productCart=new ArrayList<>();
+    public void addToTheCart(Product product){
+        productCart.add(product);
+    }
+    public void removeFromCart(String productName) {
+        for (Product pro : productCart) {
+            if (pro.getProductName().equalsIgnoreCase(productName)) ;
+            productCart.remove(pro);
+        }
+
+    }
+    public int totalNumberOfProducts(){
+        return productCart.size();
+    }
+    public void removeFromCart(Product product){
+        productCart.remove(product);
+    }
+    public BigDecimal calculatePriceInCart(){
+        BigDecimal total=BigDecimal.ZERO;
+        for(Product product:productCart){
+       total= total.add(product.getCost());
+        }
+   return total;
+    }
+    public List<Product> displayAllItems(){
+       return productCart;
+    }
+}
+```
+### Controller Package
+The last package to review is the controller, in the controller we have our end point or apis that connect to the internet all the basic functionalities of your service implementation.
+```java
+@RestController
+@RequestMapping("cart")
+public class controller {
+    @Autowired
+    DataServiceImpl dataService;
+    @Autowired
+    CartServiceImpl cartService;
+   @PostMapping("/create_customer")
+   public Customer registerCustomer(@RequestBody Customer customer){
+ return  dataService.registerCustomer(customer);
+    }
+    @GetMapping("/get_all_customers")
+    public List<Customer> gettingAllList(){
+       return dataService.gettingAllCustomers();
+    }
+
+    @PostMapping("/create_admin")
+    public Admin registerAdmin(@RequestBody Admin admin){
+        return  dataService.registerAdmin(admin);
+    }
+    @GetMapping("/get_all_admins")
+    public List<Admin> getAllAdmin(){
+        return dataService.getAllAdmin();
+    }
+    @PostMapping("/{id}/add_product")
+    public void addProducts(@PathVariable Long id ,@RequestBody Product product){
+         cartService.addProductToCartByCustomer(id,product);
+    }
+@GetMapping("/cal_price")
+    public BigDecimal calculatingPrice(){
+       return cartService.calculatePrice();
+}
+@GetMapping("/{id}/payment")
+    public BigDecimal payment(@PathVariable Long id){
+       return cartService.calculatingPurchase(id);
+}
+@GetMapping("/{id}/view_all_product")
+    public List<Product> viewAllProducts(@PathVariable Long id){
+       return cartService.checkAllTheProductsInTheCartByCustomer(id);
+}
+@GetMapping("/{id}/view_all_productAdmin")
+    public List<Product> viewAllProduct(@PathVariable Long id){
+    return cartService.checkAllTheProductsInTheCartByAdmin(id);
+    }
+}
+```
+N/B POSTMapping is for creating new objects, GetMapping, PatchMapping and PutMapping are for updating, DeleteMapping is for deleting.
```

Comment:
Refer to the annotations as `@PostMapping` `@GetMapping`, `@PatchMapping`, `@DeleteMapping`, and `@PatchMapping` for consistency.